embark-consult.el

```diff
@@ -163,11 +216,25 @@ This function is meant to be added to `embark-collect-mode-hook'."
   ("g" revert-buffer))
 
 (defun embark-consult-export-grep (lines)
-  "Create a grep mode buffer listing LINES."
+  "Wraps `grep-mode' setup and a writer of results from `consult-grep'
+type of commands. It is later associated with embark grep exporter"
+  (embark-consult--setup-grep-mode
+   "Exported grep results:\n\n"
+   lines
+   #'embark-consult--grep-results-writer))
+
+(defun embark-consult--grep-results-writer (lines)
+  "Simplistic writer to insert results produced byte
+`consult-grep' range of commands into a grep-mode buffer."
+  (dolist (line lines) (insert line "\n")))
+
+(defun embark-consult--setup-grep-mode (message lines results-writer)
```

Comment:
Do we use LINES somewhere? Otherwise we shouldn't pass it as an argument. Instead the RESULTS-WRITER can be a closure.